app/models/manageiq/providers/openstack/base_metrics_capture.rb

```diff
@@ -92,9 +93,17 @@ def list_meters(filter)
   def meter_names(metric_capture_module)
     @meter_names ||= metric_capture_module::COUNTER_NAMES
   end
+  
+  included do
+    cache_with_timeout(:counters_by_vm, 30.seconds) { Hash.new }
+  end
 
   def find_meter_counters(metric_capture_module, resource_filter, metadata_filter, log_header)
-    counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
+    counters = []
+    if self.class.counters_by_vm.dig(ems.id, target.ems_ref).nil?
+      counters = list_resource_meters(resource_filter, log_header) + list_metadata_meters(metadata_filter, log_header)
+      self.class.counters_by_vm.store_path(ems.id, target.ems_ref, counters)
```

Comment:
Hey so this looks like it is still caching the counter values rather than the meter names, this is the API call that is done every metrics capture call for every VM https://github.com/ManageIQ/manageiq-providers-openstack/blob/master/lib/manageiq/providers/openstack/legacy/openstack_handle/metric_delegate.rb#L17-L31 so I was expecting that we'd cache that.

Caching the counter values for 30 seconds doesn't really help us

Reply:
Actually its possible I'm not understanding what is actually stored in `counters`, but it shouldn't be the value of a counter since those are expected to change constantly, it should be the name of a counter which doesn't change very often if at all.